index.bs

```diff
@@ -111,19 +113,26 @@ A {{MediaStreamTrackProcessor}} internally contains a circular queue
 that allows buffering incoming media frames delivered by the track it
 is connected to. This buffering allows the {{MediaStreamTrackProcessor}}
 to temporarily hold frames waiting to be read from its associated {{ReadableStream}}.
-If the application is reading frames too slowly and the queue has reached its
-maximum size when the track delivers a new frame, the {{MediaStreamTrackProcessor}}
-will drop the oldest frame from the queue and push the new frame into the queue.
-The maximum size of the queue is decided by the UA and can change dynamically.
-The application can influence the maximum size of the queue via a parameter provided
-in the {{MediaStreamTrackProcessor}} constructor. In this case, the
-{{MediaStreamTrackProcessor}} will make sure that the size of the queue
-is never greater than the value provided by the application, but the actual maximum
-size of the queue can be smaller than such value.
-A {{MediaStreamTrackProcessor}} makes frames available to the its
+The application can influence the maximum size of the queue via a parameter
+provided in the {{MediaStreamTrackProcessor}} constructor. However, the
+maximum size of the queue is decided by the UA and can change dynamically,
+but it will not exceed the size requested by the application.
+When the application provides the maximum size parameter on the constructor,
+buffering will work such that if the application is reading frames too slowly
+and the queue has reached its maximum size when the track delivers a new frame,
+the {{MediaStreamTrackProcessor}} will drop the oldest frame from the queue
+and push the new frame into the queue.
+If the application does not provide a maximum size parameter, the UA is free
+to decide the buffering policy that will be applied by
+{{MediaStreamTrackProcessor}}. However, in all cases frames that are not dropped
```

Comment:
We cannot get rid of appMaxBufferSize, but I renamed it to maxBufferSize and got rid of the old maxBufferSize.
I rephrased it in a way that is similar, but even simpler than your proposal.
Basically, just use the regular circular queue behavior when a new frame comes, but allow the UA to remove any frames at any time.
I see no need to say that it must leave at least 1 in the queue.

Reply:
OK, I think you are correct that as currently written, we don't necessarily need to specify special behavior when maxBufferSize is 1.

However, I think it would be good to clarify that when a new frame arrives, the UA should remove older frames, if necessary, rather than dropping the new frame.